x-pack/plugin/ml/src/test/java/org/elasticsearch/xpack/ml/filestructurefinder/DelimitedFileStructureFinderTests.java

```diff
@@ -65,14 +65,57 @@ public void testCreateConfigsGivenCompleteCsv() throws Exception {
         assertEquals(Collections.singletonList("ISO8601"), structure.getJodaTimestampFormats());
     }
 
+    public void testCreateConfigsGivenIncompleteCsv() throws Exception {
+        String sample = "time,message\n" +
+            "2018-05-17T13:41:23,hello\n" +
+            "2018-05-17T13:41:25,hello\n" +
+            "2018-05-17T13:41:26,hello\n" +
+            "2018-05-17T13:41:27,hello\n" +
+            "2018-05-17T13:41:28,hello\n" +
+            "2018-05-17T13:41:29,hello\n" +
+            "2018-05-17T13:41:30,hello\n" +
+            "2018-05-17T13:41:31,hello\n" +
+            "2018-05-17T13:41:32,hello\n" +
+            "badrow\n" +
```

Comment:
Examples we have are:

1. https://github.com/elastic/kibana/issues/31065 - summary rows => inappropriate to include
2. https://github.com/elastic/kibana/issues/60196#issuecomment-599510164 - missing last column => would be good to include

So one of each 🤦 

So for this PR it's probably best that the bad rows get ignored from the analysis.  But we need to document that, as people will be losing data.  Also, what does the end-to-end experience look like, including file import?  Does the CSV ingest processor report errors for the bad rows that get propagated back to the file data visualizer?  I guess even in the long run if it's impossible to configure the CSV ingest processor to treat missing trailing columns as empty then we'll always have to ignore rows with fewer fields than the header row.

Reply:
I will verify the following, but my gut is that these should be the actions on our two scenarios:

1. missing a value in the middle: this could probably end up causing an ingest failure. I am pretty sure the UI shows what the bulk ingest failures were, and these usually say "field foo could not be mapped to type bar".

2. missing value at the end: Doc should import fully without issues as there will be no mapping conflicts. The field is just `null` or `missing` from the resulting doc. Zero data loss.

Reply:
Here is how a truly bad row is handled: 
<img width="761" alt="image" src="https://user-images.githubusercontent.com/4357155/80495661-33e36000-8936-11ea-90b4-15cbe4cc6443.png">

Here is a row that is simply missing the last field:
```
{
        "_index" : "fooo",
        "_id" : "sG0QwXEBJGToJIFIoH8F",
        "_score" : 1.0,
        "_source" : {
          "@timestamp" : "2018-05-17T13:41:35.000-04:00",
          "time" : "2018-05-17T13:41:35",
          "message" : "2018-05-17T13:41:35"
        }
      },
```

I think this is an artifact of:
```
{
      "csv": {
        "field": "message",
        "target_fields": [
          "time",
          "message"
        ],
        "ignore_missing": false
      }
    },
```
Since the header field name was the same as the source field of `message`.

Here is what it would look like when the missing column is not called `message`
Processor
```
{
      "csv": {
        "field": "message",
        "target_fields": [
          "time",
          "text"
        ],
        "ignore_missing": false
      }
    },
```
Resulting doc
```
{
        "_index" : "asdf",
        "_id" : "Pm0VwXEBJGToJIFIEoAX",
        "_score" : 1.0,
        "_source" : {
          "@timestamp" : "2018-05-17T13:41:35.000-04:00",
          "time" : "2018-05-17T13:41:35"
        }
      },
```

Reply:
OK that's interesting.  So you get past the CSV processor if you have fewer columns in a row than you have header columns.  After that the document may or may not get ingested depending on whether fields clash with subsequent processors.  So `badrow` cannot be parsed as a date, but if the first column had been a keyword field rather than a date then that would have gone through.

I think this actually isn't too bad.  If we deduce mappings based on the rows containing all the header fields, and then others get ingested or not depending on how unexpectedly different they were.